src/LayerEditor/ui/data/le_model.py

```diff
@@ -98,41 +127,94 @@ def confront_file_timestamp(self):
                 pass
         return False
 
-    def add_region(self, name, dim):
-        """ Add new region according to the provided data from the current tab and select it."""
+    def add_region(self, reg):
         with undo.group("Add new region"):
-            reg = RegionItem(name=name, dim=dim)
             self.regions_model.add_region(reg)
             self.gui_curr_block.gui_selected_layer.set_gui_selected_region(reg)
-        self.region_list_changed.emit()
-        self.invalidate_scene.emit()
+            self.emit_region_list_changed()
+            self.emit_invalidate_scene()
         return reg
 
     def delete_region(self, reg):
-        with undo.group("Add new region"):
+        with undo.group("Delete new region"):
             self.regions_model.delete_region(reg)
-            for block in self.blocks_model.blocks.values():
-                for layer in block.layers:
-                    if layer.gui_selected_region == reg:
-                        layer.set_gui_selected_region(RegionItem.none)
-        self.region_list_changed.emit()
-        self.invalidate_scene.emit()
-
-    def get_curr_layer_index(self):
-        return self.gui_curr_block.layers.index(self.gui_curr_block.gui_selected_layer)
+            self.emit_region_list_changed()
+            self.emit_invalidate_scene()
 
     def is_region_used(self, reg):
-        dim = self.regions_model.regions[reg].dim
+        """Returns True if reg is assigned to any shape in any layer in any block"""
         for block in self.blocks_model.blocks.values():
-            for layer in block.layers:
+            for layer in block.layers_dict.values():
+                dim = self.regions_model.regions[reg].dim
                 if layer.is_stratum:
                     dim -= 1
                     if dim < 0:
                         continue
```

Comment:
That doesn't apply.  This method checks if region is used. If I check region with dim==3 and Im checking in fracture layer I just want to skip because no fracture layer cannot have region with dim==3.

Reply:
move dim = self.regions_model.regions[reg].dim outside of loop and make new temp dim